.gitlab-ci.yml

```diff
@@ -24,6 +24,7 @@ stages:
   BUILD_HIP: "OFF"
   BUILD_HWLOC: "ON"
   FAST_TESTS: "OFF"
+  MIXED_PRECISION: "ON"
```

Comment:
Can we do it the other way ? Switch it `ON` only for a few jobs and `OFF` by default ? I don't see any particular need to have it on for all jobs.

Reply:
The reason I chose this default is that `MIXED_PRECISION` has two impacts:
1. it limits the number of kernels that will be instantiated to a smaller number
2. it changes the `core` implementation to use only the same-precision kernels.
 
To test that everything works correctly, we need to check that both parts work correctly. If the device kernel compile with `MIXED_PRECISION=ON`, then they will also compile with `OFF`, and the kernels might have their own mixed-precision quirks, so I would like to test them on different platforms.

When testing with `MIXED_PRECISION=OFF`, we only test that `make_temporary_conversion` and the non-mixed precision kernels work correctly. Since this behavior is already tested elsewhere, I wanted the focus to be on mixed-precision kernels, with only one instantiation of the conversion-based mode for each compiler vendor to be sure it works. I should do the same for Windows and MacOS, though!